core/src/main/java/org/bitcoinj/core/Peer.java

```diff
@@ -451,6 +459,12 @@ protected void processMessage(Message m) throws Exception {
             currentFilteredBlock = null;
         }
 
+        // No further communication is possible until version handshake is complete.
+        if (!(m instanceof VersionMessage || m instanceof VersionAck
+                || (versionHandshakeFuture.isDone() && !versionHandshakeFuture.isCancelled())))
```

Comment:
Yes, that's how I understand the [documentation of Futures.allAsList()](http://docs.guava-libraries.googlecode.com/git/javadoc/com/google/common/util/concurrent/Futures.html#allAsList%28com.google.common.util.concurrent.ListenableFuture...%29).